### PR TITLE
Remove jobs from sig-release-<release_version>-all if they're also in -blocking or -informing

### PR DESCRIPTION
**Pull request by mariantalla**
This PR covers all jobs under sig-release dashboards that I could find; for now it just removes them if the existing tooling has already placed them in another sig-release dashboard. If jobs are monitored in other (not sig-release monitored)dashboards, this PR keeps them under sig-release-<release_version>-all for now (in practice I don't remember seeing of any such jobs).

xref: https://github.com/kubernetes/test-infra/issues/11977

/cc @spiffxp @alejandrox1 @Katharine

### CODE REVIEW

File: experiment/generate_tests.py [hunk 1]
```diff
@@ -268,6 +268,14 @@ def __get_testgrid_config(self):
...
         return tg_config
 
+    def initialize_dashboards_with_release_blocking_info(self, version):
+        dashboards = []
+        if self.job.get('releaseBlocking'):
+            dashboards.append('sig-release-%s-blocking' % version)
+        else:
+            dashboards.append('sig-release-%s-all' % version)
+        return dashboards
+
```
Comment (alejandrox1) [inline experiment/generate_tests.py]: Love this bit!

File: experiment/generate_tests_test.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+import tempfile
+import shutil
+from generate_tests import E2ETest
+
+class TestGenerateTests(unittest.TestCase):
```
Comment (smourapina) [inline experiment/generate_tests_test.py]: Very nice to see tests added here!

**Comment by Katharine** [comment]
If you want this to represent a policy change going forward, you'll also want to do something more nuanced in the [config-forker](https://github.com/kubernetes/test-infra/blob/master/experiment/config-forker/main.go#L267).

File: experiment/generate_tests_test.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        }
+
+    def tearDown(self):
+        shutil.rmtree(self.temp_directory)
+
+    def test_e2etests_testgrid_annotations_default(self):
+        generator = E2ETest(self.temp_directory, self.job_name, self.job, self.config)
+        _, prow_config, _ = generator.generate()
+        dashboards = prow_config["annotations"]["testgrid-dashboards"]
+        assert "sig-release-2.4-blocking" not in dashboards
```
Comment (Katharine) [inline experiment/generate_tests_test.py]:
```suggestion
        self.assertFalse("sig-release-2.4-blocking" in dashboards)
```

File: experiment/generate_tests_test.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+        assert "sig-release-2.4-all" in dashboards
```
Comment (Katharine) [inline experiment/generate_tests_test.py]:
```suggestion
        self.assertTrue("sig-release-2.4-all" in dashboards)
```

File: experiment/generate_tests_test.py [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    def test_e2etests_testgrid_annotations_blocking_job(self):
+        self.job = {
+            "releaseBlocking": True,
+            "interval": "1h"
+        }
+
+        generator = E2ETest(self.temp_directory, self.job_name, self.job, self.config)
+        _, prow_config, _ = generator.generate()
+        dashboards = prow_config["annotations"]["testgrid-dashboards"]
+        assert "sig-release-2.4-blocking" in dashboards
```
Comment (Katharine) [inline experiment/generate_tests_test.py]:
```suggestion
        self.assertTrue("sig-release-2.4-blocking" in dashboards)
```

File: experiment/generate_tests_test.py [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+        assert "sig-release-2.4-all" not in dashboards
```
Comment (Katharine) [inline experiment/generate_tests_test.py]:
```suggestion
        self.assertFalse("sig-release-2.4-all" in dashboards)
```

File: experiment/generate_tests_test.py (hunk 3 above)
Comment (mariantalla) [reply experiment/generate_tests_test.py]: Makes sense, thanks 👍

**Comment by mariantalla** [comment]
> If you want this to represent a policy change going forward, you'll also want to do something more nuanced in the [config-forker](https://github.com/kubernetes/test-infra/blob/master/experiment/config-forker/main.go#L267).

Thanks, I missed that one. Yep, I think it makes sense to carry forward. I'll make the change in a separate PR though.

**Comment by spiffxp** [comment]
Followup issue for the config-forker changes: https://github.com/kubernetes/test-infra/issues/12848

**Pull request merged**